test/dynamo/test_export.py

```diff
@@ -2264,6 +2264,22 @@ def my_dyn_fn(a, b, c):
         constraints.append(dynamic_dim(z, 0) == dynamic_dim(x, 0))
         torch._dynamo.export(my_dyn_fn, constraints=constraints)(x, y, z)
 
+    def test_remove_redundant_dynamic_dim_in_error_message(self):
+        def foo(x, y):
+            if x.shape[0] == y["k"].shape[0]:
+                return x + 1
+            else:
+                return x - 1
+
+        a = torch.randn(3)
+        b = torch.randn(3)
+        with self.assertRaisesRegex(
+            torch._dynamo.exc.UserError, "\\[\n.*\n.*dynamic_dim.*==.*dynamic_dim.*\n.*\\]"
```

Comment:
Optional nit:

This regex is overly specific, and it can become a "change detector". A better way is to detect that there is no patterns like "dynamic_dim(t, idx)," on its own line.